example/newreposecret/main.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// newreposecret creates a new secret in GitHub for a given owner/repo
+// It has two required flags for owner and repo, and takes in one argument for the name of the secret to add.
+// Provide the value of the secret you want to add with an environment variable of the same name.
+// To authenticate with GitHub provide it via an environment variable GITHUB_AUTH_TOKEN
+//
+// To verify the new secret navigate to GitHub Repository > Settings > left side options bar > Secrets
+//
+// Usage:
+//	export GITHUB_AUTH_TOKEN=<auth token from github that has secret create rights>
+//	export SECRET_VARIABLE=<secret value of the secret variable>
+//	go run main.go -owner <owner name> -repo <repository name> SECRET_VARIABLE
+//
+// Example:
+//	export GITHUB_AUTH_TOKEN=0000000000000000
+//	export SECRET_VARIABLE="my-secret"
+//	go run main.go -owner google -repo go-github SECRET_VARIABLE
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	sodium "github.com/GoKillers/libsodium-go/cryptobox"
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+	"log"
+	"os"
+)
+
+var (
+	repo  = flag.String("repo", "", "The repo that the secret should be added to, ex. go-github")
+	owner = flag.String("owner", "", "The owner of there repo this should be added to, ex. google")
+)
+
+func main() {
+	flag.Parse()
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token == "" {
+		log.Fatal("please provide a GitHub API token via env variable GITHUB_AUTH_TOKEN")
+	}
+
+	if *repo == "" {
+		log.Fatal("please provide required flag --repo to specify GitHub repository ")
+	}
+
+	if *owner == "" {
+		log.Fatal("please provide required flag --owner to speify GitHub user/org owner")
+	}
+
+	secretName, err := getSecretName()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	secretValue, err := getSecretValue(secretName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ctx, client, err := githubAuth(token)
+	if err != nil {
+		log.Fatalf("unable to authorize using env GITHUB_AUTH_TOKEN: %v", err)
+	}
+
+	err = addRepoSecret(ctx, client, *owner, *repo, secretName, secretValue)
+	if err != nil {
```

Comment:
this line can be removed (after adding it to the previous line).

Reply:
updated!